add MISMIP test config

justinh2002 opened the pull request:
---
:rocket: The latest prerelease `access-issm/pr17-12` at d3a5e1f5a4c701ac73ef5b99b74aa09db85d3a58 is here: https://github.com/ACCESS-NRI/ACCESS-ISSM/pull/17#issuecomment-3408897331 :rocket:

justinh2002 committed:
modified README for MDR

justinh2002 committed:
modified README.md [no ci]

justinh2002 committed:
Update README.md

Co-authored-by: Aidan Heerdegen <aidan.heerdegen@gmail.com>

justinh2002 committed:
Update README.md

Co-authored-by: Aidan Heerdegen <aidan.heerdegen@gmail.com>

justinh2002 committed:
Update README.md

Co-authored-by: Aidan Heerdegen <aidan.heerdegen@gmail.com>

justinh2002 committed:
Update README.md

Co-authored-by: Aidan Heerdegen <aidan.heerdegen@gmail.com>

justinh2002 committed:
Update README.md

Co-authored-by: Aidan Heerdegen <aidan.heerdegen@gmail.com>

justinh2002 committed:
Update README.md

Co-authored-by: Aidan Heerdegen <aidan.heerdegen@gmail.com>

justinh2002 committed:
Update README.md

Co-authored-by: Aidan Heerdegen <aidan.heerdegen@gmail.com>

justinh2002 committed:
Update README.md

Co-authored-by: Aidan Heerdegen <aidan.heerdegen@gmail.com>

justinh2002 committed:
Update README.md

Co-authored-by: Aidan Heerdegen <aidan.heerdegen@gmail.com>

justinh2002 committed:
removed mention of CONTRIBUTING.md

justinh2002 committed:
Update README.md

Co-authored-by: Aidan Heerdegen <aidan.heerdegen@gmail.com>

justinh2002 committed:
removed line about access-hive docs

justinh2002 committed:
Merge branch '2_mdr_readme' of github.com:ACCESS-NRI/ACCESS-ISSM into 2_mdr_readme

justinh2002 committed:
removed line on access-hive docs

justinh2002 committed:
Update README.md

Co-authored-by: Aidan Heerdegen <aidan.heerdegen@gmail.com>

justinh2002 committed:
Update README.md

Co-authored-by: Aidan Heerdegen <aidan.heerdegen@gmail.com>

justinh2002 committed:
add +ad variant install

justinh2002 committed:
Merge branch 'main' into justin/ad

justinh2002 committed:
changed release tag

justinh2002 committed:
Merge branch 'justin/ad' of github.com:ACCESS-NRI/ACCESS-ISSM into justin/ad

justinh2002 committed:
version modified

justinh2002 committed:
modify projections

justinh2002 committed:
changed version json for spack-packages

justinh2002 committed:
add new git tag

justinh2002 committed:
change branch to test branch of spack

justinh2002 committed:
amend to test branch

justinh2002 committed:
fix commit versioning

justinh2002 committed:
update spack-packages to point to PR branch

justinh2002 committed:
added test directory and setup files

justinh2002 committed:
reorganise

CodeGat reviewed (changes requested):
Spack stuff looks good, but the version of `spack-packages` needs to be a tag rather than a branch for a proper release

lawrenceabird commented on test/Mismip.py:
It would be useful to add some docstring to the top of this file to explain what it is, how it is structured, and how people should use it. Note that it relies on the ISSM install for the module imports etc.

justinh2002 replied:
done

lawrenceabird commented on test/Mismip.py:
Add some comments throughout. Perhaps where there are print statements and there are natural breaks/code blocks, add a small comment block to explain what each step is doing.

justinh2002 replied:
done

lawrenceabird commented on test/mismip_driver.py:
As with `Mismip.py`, add a docstring/header to the file to explain what it does, how it's structured, and how it should be used etc. There is *a lot* in here, so it would be beneficial to provide some signposting upfront to help users navigate it.

Note required ISSM imports etc.

lawrenceabird commented on test/mismip_driver.py:
```diff
@@ -0,0 +1,780 @@
+     'trunkprefix',  '34;47;2'
+)
+# ------------------------------------------------------------------------------
+#  STEP1: Mesh_generation
```
Expand on the comment blocks/headers for each step. Give some more details about what is actually happening in each section/perform call. What model is being loaded/what step does it depend on?

Add some intermediate comments at individual code chunks within these larger headings.

justinh2002 replied:
done

lawrenceabird commented on test/run_mismip_first.sh:
```diff
@@ -0,0 +1,57 @@
+# Runs the MISMIP workflow just far enough to generate the
+# Glen–vs‑ESTAR comparison plots, skipping the later “enhanced”
+# or full‑Stokes experiments.
+#
```
Make this file header a little more verbose. Are these steps that are defined here? Perhaps consider numbering them based on corresponding sections below.

lawrenceabird commented on test/run_pps.sh:
Add file header/docstring to explain what this script does and how it should be used. Make comments throughout a little more verbose so it's clear exactly whats going on.

justinh2002 replied:
done

lawrenceabird reviewed (changes requested):
There's *a lot* in here! I've added some comments to the MISMIP-related files. Particularly, the files would benefit from some detailed headers/docstrings and some more verbose and detailed comments throughout. Assume these are the first ISSM-related scripts that users have seen, so they need to be walked through carefully.

justinh2002 committed:
Merge branch 'main' into justinh2002/configuration

justinh2002 committed:
Correct model name in README

Updated model name from 'ACCESS Ice Sheet System Model' to 'ACCESS Ice Sheet Sea-Level Model' in the README.

justinh2002 committed:
added docstrings and annotation

justinh2002 committed:
Merge branch 'justinh2002/configuration' of github.com:ACCESS-NRI/ACCESS-ISSM into justinh2002/configuration

justinh2002 committed:
add annotations

justinh2002 committed:
add comments and headers

justinh2002 committed:
add comments and annotations

justinh2002 committed:
add versions json update

justinh2002 commented on config/versions.json:
```diff
@@ -1,5 +1,5 @@
     "$schema": "https://raw.githubusercontent.com/ACCESS-NRI/schema/main/au.org.access-nri/model/deployment/config/versions/3-0-0.json",
     "spack": "0.22",
-    "spack-packages": "2025.04.001"
+    "spack-packages": "justin/ad_mumps"
```
reverted

justinh2002 committed:
i dont want ad

justinh2002 committed:
Added Domain config

justinh2002 commented:
!redeploy

justinh2002 committed:
added front.exp

justinh2002 committed:
change rheology law

CodeGat commented:
You're really testing the concurrent deployment corner case here @justinh2002, thanks :laughing:
